### PR TITLE
Test theory modal

### PR DESCRIPTION
**Pull request by lebedevaanastasiya**

### CODE REVIEW

File: src/app/services/test-themes.service.ts [hunk 1]
```diff
@@ -24,4 +24,8 @@ export class TestThemesService {
   remove(themeId) {
     return this.http.delete(`${this.config.API}/test-topics/${themeId}`, {});
   }
+
+  updateTherory(themeId, theory) {
```
Comment (mbarouski) [inline src/app/services/test-themes.service.ts]: `updateTheory`

File: src/app/components/admin-test-theme.component.ts [hunk 2]
```diff
@@ -64,6 +71,41 @@ export class AdminTestThemeComponent implements OnInit {
...
+        this.alertService.showErrorMessage(error_message);
+      });
+  }
+
+  onTheoryFormClose() {
+    this.closeTheoryModal();
+  }
+
+  restoreState(theme) {
+    this.closeTheoryModal();
```
Comment (mbarouski) [inline src/app/components/admin-test-theme.component.ts:105]: why we need to use it

File: src/app/components/admin-test-theme.component.ts [hunk 3]
```diff
@@ -64,6 +71,41 @@ export class AdminTestThemeComponent implements OnInit {
...
+    this.theme = theme;
```
Comment (mbarouski) [inline src/app/components/admin-test-theme.component.ts:106]: there is no currentTheory setting

**Review by mbarouski — CHANGES_REQUESTED** [review]

File: src/app/components/admin-test-theme.component.ts (hunk 2 above)
Comment (lebedevaanastasiya) [reply src/app/components/admin-test-theme.component.ts:105]: You mean this restoreState function?

Comment (mbarouski) [reply src/app/components/admin-test-theme.component.ts:105]: yes

Comment (lebedevaanastasiya) [reply src/app/components/admin-test-theme.component.ts:105]: We don't need

File: src/app/components/admin-test-theme.component.ts (hunk 3 above)
Comment (lebedevaanastasiya) [reply src/app/components/admin-test-theme.component.ts:106]: Why do I need it here?

Comment (mbarouski) [reply src/app/components/admin-test-theme.component.ts:106]: line 52

File: src/app/components/admin-test-theme.component.ts (hunk 2 above)
Comment (mbarouski) [reply src/app/components/admin-test-theme.component.ts:105]: so, why is it here?

**Pull request merged**